app/jobs/mail_job.rb

```diff
@@ -0,0 +1,13 @@
+class MailJob < Que::Job
+  def run
+    return if queued?
+
+    NudgeMail.new.call
+  end
+
+private
+
+  def queued?
+    Que.job_stats.any? { |job| job[:job_class] == 'MailJob' && job[:count] > 1 }
```

Comment:
HMW get the same check for our dashboard?

Reply:
I was looking at a way to stop duplicate jobs being added to the queue in the first place, but wasn't able to make much progress. Instead, I've added a helper to guard against duplicate jobs in the run method of the jobs [dbbefa7](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/726/commits/dbbefa716e204407f0ff65666d890aa8a516e67c)